netket_fidelity/operator/singlequbit_gates.py

```diff
@@ -1,30 +1,16 @@
 from functools import partial
-
 import numpy as np
-
 import jax
 import jax.numpy as jnp
-
 from jax.tree_util import register_pytree_node_class
-
-from netket.hilbert import Spin
 from netket.operator import DiscreteJaxOperator, spin
 
 
 @register_pytree_node_class
 class Rx(DiscreteJaxOperator):
     def __init__(self, hi, idx, angle):
-        if not isinstance(hi, Spin):
-            raise TypeError(
-                """The Hilbert space used by Rx must be a `Spin` space.
-
-                This limitation could be lifted by 'fixing' the method
-                `get_conn_and_mels` to work with arbitrary hilbert spaces, which
-                should be relatively straightforward to do, but we have not done so
-                yet.
-                """
-            )
         super().__init__(hi)
+        self._local_states = hi.local_states
```

Comment:
Cast it to jax array so jax does not have to do it at every function call

```suggestion
        self._local_states = jnp.asarray(hi.local_states)
```

Reply:
should also be done for the other arguments